pyomo/contrib/solver/solvers/ipopt.py

```diff
@@ -499,46 +535,136 @@ def solve(self, model, **kwds):
+                    else:
+                        iter_data["step_acceptance"] = None
+
+                    # Attempt to cast all values to float where possible
+                    for key in iter_data:
+                        try:
+                            if iter_data[key] == '-':
+                                iter_data[key] = None
+                                continue
+                            if isinstance(iter_data[key], bool):
+                                continue
+                            iter_data[key] = float(iter_data[key])
+                        except (ValueError, TypeError):
+                            pass
+
+                    all_iterations.append(iter_data)
+
+            parsed_data['iteration_log'] = all_iterations
+
+        # Extract scaled and unscaled table
+        scaled_unscaled_match = re.findall(
+            r'Objective\.\.+:\s+([-+eE0-9.]+)\s+([-+eE0-9.]+).*?'
+            r'Dual infeasibility\.\.+:\s+([-+eE0-9.]+)\s+([-+eE0-9.]+).*?'
+            r'Constraint violation\.\.+:\s+([-+eE0-9.]+)\s+([-+eE0-9.]+).*?'
+            r'Complementarity\.\.+:\s+([-+eE0-9.]+)\s+([-+eE0-9.]+).*?'
+            r'Overall NLP error\.\.+:\s+([-+eE0-9.]+)\s+([-+eE0-9.]+)',
+            output,
+            re.DOTALL,
+        )
+        if scaled_unscaled_match:
+            scaled = {
+                "incumbent_objective": float(scaled_unscaled_match[0][0]),
+                "dual_infeasibility": float(scaled_unscaled_match[0][2]),
+                "constraint_violation": float(scaled_unscaled_match[0][4]),
+                "complementarity_error": float(scaled_unscaled_match[0][6]),
+                "overall_nlp_error": float(scaled_unscaled_match[0][8]),
+            }
+            unscaled = {
+                "incumbent_objective": float(scaled_unscaled_match[0][1]),
+                "dual_infeasibility": float(scaled_unscaled_match[0][3]),
+                "constraint_violation": float(scaled_unscaled_match[0][5]),
+                "complementarity": float(scaled_unscaled_match[0][7]),
+                "overall_nlp_error": float(scaled_unscaled_match[0][9]),
+            }
+
+            parsed_data.update(unscaled)
+            parsed_data['final_scaled_results'] = scaled
+
+        # Newer versions of IPOPT no longer separate timing into
+        # two different values. This is so we have compatibility with
+        # both new and old versions
+        cpu_time = re.findall(r'Total CPU secs in .*? =\s+([0-9.]+)', output)
+        ipopt_seconds_match = re.search(
+            r'Total seconds in IPOPT\s+=\s+([0-9.]+)', output
+        )
+        if cpu_time and len(cpu_time) >= 2:
+            parsed_data['cpu_seconds'] = {
+                'nofunc_time': float(cpu_time[0]),
+                'func_time': float(cpu_time[1]),
+            }
+        elif ipopt_seconds_match:
+            parsed_data['cpu_seconds'] = {
+                'total_time': float(ipopt_seconds_match.group(1))
+            }
```

Comment:
Would the following be easier:
```python
parsed_data['cpu_seconds'] = {
    k.strip(): float(v) for k, v in 
    re.findall(r'Total(?: CPU)? sec(?:ond)?s in ([^=]+)=\s*([0-9.]+)')
}
```

Reply:
It would make `cpu_seconds` look like this (ipopt 3.13):
```
{'IPOPT (w/o function evaluations)': 0.001, 'NLP function evaluations': 0.0}
```
Which is fine but then I have to do more logic in the actual `solve` to convert from this dict to the timer on the `Results` object. I'd prefer to keep it as it is because it's more clear.

Reply:
FWIW, as someone who has worked with Ipopt, I prefer `IPOPT (w/o function evaluations)` (et al.):  It has the advantages of being explanatory and familiar (we are not injecting our own terminology unnecessarily).